### PR TITLE
Add manual wallet chain switch

### PR DESCRIPTION
**Pull request by ianrowan**
## Link to Issue
Closes: #6530 

## Description of Changes
- adds `switchNetwork()` method to metamask and coin base web wallets. 
- method can be called to check if on network, automatically switch if not, and add if not in metamask(see below) 

## Other Considerations

- I noticed that in out normal `enable()` flow if a network is not in the users wallet, we add it with our own node/rpc. This is sub-optimal b/c the user will essentially carry our rpc around forever and make their transactions through it vs using a public node like wallets usually do
    - I added code to the new method to use public chainlist rpcs for this and we should consider the same for enable() but wanted more input here: CC @jnaviask

### CODE REVIEW

**Comment by ianrowan** [comment]
@masvelio You can use the new wallet method here to switch chains in metamask before first community stake transactions, etc

**Comment by masvelio** [comment]
@ianrowan I am not 100% where should I use it and if this is necessary to do before the launch of community stake?

File: packages/commonwealth/client/scripts/controllers/app/webWallets/coinbase_web_wallet.ts [hunk 1]
```diff
@@ -198,6 +198,49 @@ class CoinbaseWebWalletController implements IWebWallet<string> {
...
+          });
+        } catch (error) {
+          if (error.code === 4902) {
+            const chains = await $.getJSON(
+              'https://chainid.network/chains.json',
+            );
+            const baseChain = chains.find((c) => c.chainId === communityChain);
+            // Check if the string contains '${' and '}'
+            const rpcUrl = baseChain.rpc.filter((r) => !/\${.*?}/.test(r));
+            const url =
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/controllers/app/webWallets/coinbase_web_wallet.ts:222]: 'url' is assigned a value but never used

File: packages/commonwealth/client/scripts/controllers/app/webWallets/metamask_web_wallet.ts [hunk 2]
```diff
@@ -209,6 +209,49 @@ class MetamaskWebWalletController implements IWebWallet<string> {
...
+          });
+        } catch (error) {
+          if (error.code === 4902) {
+            const chains = await $.getJSON(
+              'https://chainid.network/chains.json',
+            );
+            const baseChain = chains.find((c) => c.chainId === communityChain);
+            // Check if the string contains '${' and '}'
+            const rpcUrl = baseChain.rpc.filter((r) => !/\${.*?}/.test(r));
+            const url =
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/controllers/app/webWallets/metamask_web_wallet.ts:232]: 'url' is assigned a value but never used

**Review by masvelio — APPROVED** [review]

**Comment by mhagel** [comment]
>I noticed that in out normal enable() flow if a network is not in the users wallet, we add it with our own node/rpc. This is sub-optimal b/c the user will essentially carry our rpc around forever and make their transactions through it vs using a public node like wallets usually do

Good call-out.

**Review by mhagel — APPROVED** [review]
LGTM

**Pull request merged**